src/ucp/proto/proto_multi.c

```diff
@@ -242,17 +237,44 @@ ucs_status_t ucp_proto_multi_init(const ucp_proto_multi_init_params_t *params,
     return status;
 }
 
+size_t
+ucp_proto_multi_priv_size(const ucp_proto_multi_priv_t *mpriv, size_t full_size)
+{
+    ucs_assert(full_size >= sizeof(ucp_proto_multi_priv_t));
```

Comment:
looks hacky, is it possible to calculate just `mpriv->num_lanes*ucs_field_sizeof(ucp_proto_multi_priv_t, lanes[0])` and the rest calculate separatly in each place?